src/processes.py

```diff
@@ -94,4 +95,8 @@ def onMessage(self, message: 'Message'):
             is expected to be one of the sub-classes of the Message class
             in messages.py
         '''
+        if message.action == Action.Heartbeat and message.type == MsgType.Request:
+            #TODO: update liveness status in global dictionary
+            print(self.node.get_name(), ": I am live!!!")
```

Comment:
please use the log API to log messages.

also, this logic doesn't seem to reply back to the heartbeat. From what I can tell, it's logging it received a heartbeat instead of sending an ack.

@nishantrai18 I think we need to understand the Message and Action class better. @AndrejSafundzic and I also had some issues figuring it out.

Reply:
I am waiting to discuss who will take action of overall liveness exception. Is the publisher of heartbeat or the leader? If leader takes action, then there are no need for message to send back to the publisher, instead we can just update global dictionary of liveness status.